DQM/CTPPS/plugins/CTPPSPixelDQMSource.cc

```diff
@@ -0,0 +1,462 @@
+/******************************************
+*
+* This is a part of CTPPSDQM software.
+* Authors:
+*   F.Ferro INFN Genova
+*   Vladimir Popov (vladimir.popov@cern.ch)
+*
+*******************************************/
+
+#include "FWCore/Framework/interface/ESHandle.h"
+#include "FWCore/Framework/interface/MakerMacros.h"
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/Framework/interface/EventSetup.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "FWCore/Utilities/interface/InputTag.h"
+
+#include "DQMServices/Core/interface/DQMEDAnalyzer.h"
+#include "DQMServices/Core/interface/DQMStore.h"
+#include "DQMServices/Core/interface/MonitorElement.h"
+
+#include "DataFormats/Common/interface/DetSetVector.h"
+
+#include "DataFormats/CTPPSDetId/interface/CTPPSDetId.h"
+#include "DataFormats/CTPPSDigi/interface/CTPPSPixelDigi.h"
+#include "CondFormats/CTPPSReadoutObjects/interface/CTPPSPixelIndices.h"
+
+#include <string>
+
+//-----------------------------------------------------------------------------
+ 
+class CTPPSPixelDQMSource: public DQMEDAnalyzer
+{
+ public:
+   CTPPSPixelDQMSource(const edm::ParameterSet& ps);
+   virtual ~CTPPSPixelDQMSource();
+  
+ protected:
+   void dqmBeginRun(edm::Run const &, edm::EventSetup const &) override;
+   void bookHistograms(DQMStore::IBooker &, edm::Run const &, edm::EventSetup const &) override;
+   void analyze(edm::Event const& e, edm::EventSetup const& eSetup);
+   void beginLuminosityBlock(edm::LuminosityBlock const& lumi, edm::EventSetup const& eSetup);
+   void endLuminosityBlock(edm::LuminosityBlock const& lumi, edm::EventSetup const& eSetup);
+   void endRun(edm::Run const& run, edm::EventSetup const& eSetup);
+
+ private:
+   unsigned int verbosity;
+
+  edm::EDGetTokenT<edm::DetSetVector<CTPPSPixelDigi>> tokenDigi;
+//  edm::EDGetTokenT< edm::DetSetVector<CTPPSPixelCluster> > tokenCluster;
+
+ #define NArms 2
+ #define NStationMAX 3  // in an arm
```

Comment:
better if these defines can be const expr too [better that all of these was in one central place..]